### PR TITLE
feat: add temperature swarm plots

### PR DESCRIPTION
**Pull request by MrtinoRG**
{095DC2C7-FB11-4355-94CD-E8AEBCB41CB5}

The x-axis spines need to be solved... But as a first draft I think it's cool. Also, it is possible to see that the tendency is correct since Temp=1 results are never really higher than for Temp=0

## Summary by Sourcery

Add temperature swarm plots to visualize model performance at different temperature settings and update model naming conventions to reflect temperature information.

New Features:
- Introduce temperature swarm plots to visualize model performance at different temperature settings.

Enhancements:
- Modify the model naming convention to include temperature information, appending '-T-one' for models with temperature 1.

### CODE REVIEW

**Comment by AdrianM0** [comment]
> {095DC2C7-FB11-4355-94CD-E8AEBCB41CB5}
> 
> The x-axis spines need to be solved... But as a first draft I think it's cool. Also, it is possible to see that the tendency is correct since Temp=1 results are never really higher than for Temp=0
> 
> ## Summary by Sourcery
> Add temperature swarm plots to visualize model performance at different temperature settings and update model naming conventions to reflect temperature information.
> 
> New Features:
> 
> * Introduce temperature swarm plots to visualize model performance at different temperature settings.
> 
> Enhancements:
> 
> * Modify the model naming convention to include temperature information, appending '-T-one' for models with temperature 1.

is that the final plot version @MrtinoRG ?

File: src/scripts/plot_temperature_diffs.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    # Place the legend outside the plot to the right
+    plt.legend(loc="center left", bbox_to_anchor=(1, 0.5), fontsize="small")
+
+    # Adjust axis limits to ensure all data points are fully visible
+    ax.set_xlim(-0.05, len(df_melted["Temperature"].unique()) - 0.0001)
+    ax.set_ylim(df_melted["Score"].min() - 0.01, df_melted["Score"].max() + 0.01)
+    # Adjust the x-axis line to go only until the last label
+# Adjust the x-axis line to go only until the last label
+    ax.spines['bottom'].set_position(('outward', 0.01))
+
```
Comment (kjappelbaum) [inline src/scripts/plot_temperature_diffs.py:78]: I think the x axis might be "fixed" using `range_frame`. Something like 

```
range_frame(ax, np.array([-1,1]), df_melted['Score'].values) 
```

which should then also replace the `set_xlim` and `set_ylim` calls

**Comment by kjappelbaum** [comment]
Very cool addition! Good to see this and good for the literature to now have a citable report of this observation :)

File: src/scripts/plot_temperature_diffs.py (hunk 1 above)
Comment (kjappelbaum) [reply src/scripts/plot_temperature_diffs.py:78]: i'll handle that after merging :)

**Comment by kjappelbaum** [comment]
and actually made me aware of a potential place where we can introduce bugs due to the model namign

File: src/scripts/color_palette.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "Chemist": "#F7DC6F",
-    "o1": "#AB8D9E",
+    "GPT-4o1": "#AB8D9E",
```
Comment (kjappelbaum) [inline src/scripts/color_palette.json:3]: I think the official name really is o1?

Comment (kjappelbaum) [reply src/scripts/color_palette.json:3]: I think they tried to break with the GPT naming to indicate a big difference

Comment (MrtinoRG) [reply src/scripts/color_palette.json:3]: Yes, or OpenAI O1. I do not know when and how I changed this

Comment (MrtinoRG) [reply src/scripts/color_palette.json:3]: So, yeah I agree that this should be reverted

**Pull request merged**